cuda_core/cuda/core/experimental/_kernel_arg_handler.pyx

```diff
@@ -182,7 +182,7 @@ cdef class ParamHolder:
         for i, arg in enumerate(kernel_args):
             if isinstance(arg, Buffer):
                 # we need the address of where the actual buffer address is stored
-                self.data_addresses[i] = <void*><intptr_t>(arg._ptr.getPtr())
+                self.data_addresses[i] = <void*><intptr_t>(arg.handle.getPtr())
```

Comment:
I could easily change this to `arg._mnff.ptr`. Generally I try to avoid accessing private members from outside their class, but given that _memory.py and _kernel_arg_handler.pyx live in the same directory, it seems OK to me. (I'd be much more reluctant if the private access was from "further away", e.g. another directory, or even another repo.)

Please let me know what you prefer.